src/odb_loose.c

```diff
@@ -352,7 +355,7 @@ static int read_loose(git_rawobj *out, git_buf *loc)
 	if ((error = git_futils_readbuffer(&obj, loc->ptr)) < 0)
 		goto done;
 
-	if (!is_zlib_compressed_data((unsigned char *)obj.ptr))
+	if (!is_zlib_compressed_data((unsigned char *)obj.ptr, obj.size))
 		error = read_loose_packlike(out, &obj);
 	else
 		error = read_loose_standard(out, &obj);
```

Comment:
Huh. Why should we treat an object as packlike though when `obj.size < 2`? Shouldn't we error out in that case instead? Something like:

```
if (is_zlib_compressed_data(&is_compressed, obj.ptr. obj.size) < 0)
    return -1;

if (!is_compressed)
    packlike();
else
    standard();
```